controller/scheduler/k8s.py

```diff
@@ -0,0 +1,647 @@
+            raise RuntimeError(errmsg)
+        create = False
+        for _ in xrange(30):
+            if not create and self._get_rc_status(name) == 404:
+                time.sleep(1)
+                continue
+            create = True
+            rc = self._get_rc_(name)
+            if ("observedGeneration" in rc["status"]
+                    and rc["metadata"]["generation"] == rc["status"]["observedGeneration"]):
+                break
+            time.sleep(1)
+        return json.loads(data)
+
+    def create(self, name, image, command, **kwargs):
+        self._create_rc(name, image, command, **kwargs)
+        app_type = name.split(".")[1]
+        name = name.replace(".", "-")
+        name = name.replace("_", "-")
+        app_name = kwargs.get('aname', {})
+        route_type = kwargs.get('route_type', {})
+        if app_type == route_type:
+            try:
+                self._create_service(name, app_name, app_type)
+            except Exception as e:
+                self._delete_rc(name)
+                err = '{} (create): {}'.format(name, e)
+                raise RuntimeError(err)
+
+    def _create_service(self, name, app_name, app_type):
+        actual_pod = {}
+        for _ in xrange(300):
+            status, data, reason = self._get_pods()
+            parsed_json = json.loads(data)
+            for pod in parsed_json['items']:
+                if('generateName' in pod['metadata'] and
+                   pod['metadata']['generateName'] == name+'-'):
+                    actual_pod = pod
+                    break
+            if actual_pod and actual_pod['status']['phase'] == 'Running':
+                break
+            time.sleep(1)
+        container_id = actual_pod['status']['containerStatuses'][0]['containerID'].split("//")[1]
+        ip = actual_pod['status']['hostIP']
+        docker_cli = Client("tcp://{}:2375".format(ip), timeout=1200, version='1.17')
+        container = docker_cli.inspect_container(container_id)
+        port = int(container['Config']['ExposedPorts'].keys()[0].split("/")[0])
+        l = {}
+        l["version"] = self.apiversion
+        l["label"] = app_name
+        l["port"] = port
+        l['type'] = app_type
+        random.seed(app_name)
+        app_id = random.randint(1, 100000)
+        l["name"] = "app-"+str(app_id)
+        template = string.Template(SERVICE_TEMPLATE).substitute(l)
+        headers = {'Content-Type': 'application/json'}
+        conn_serv = httplib.HTTPConnection(self.target+":"+self.port)
+        conn_serv.request('POST', '/api/'+self.apiversion+'/namespaces/default/services',
+                          headers=headers, body=copy.deepcopy(template))
+        resp = conn_serv.getresponse()
+        data = resp.read()
+        reason = resp.reason
+        status = resp.status
+        conn_serv.close()
+        if not 200 <= status <= 299:
+            errmsg = "Failed to create Service:{} {} {} - {}".format(
```

Comment:
@kmala it "feels" weird not be able to reuse the service that already exists. 
Not reusing the service means that the assigned IP is going to be different so if some user is using the env variables to get the host and port (following the [environment-guide](https://github.com/GoogleCloudPlatform/kubernetes/tree/master/docs/user-guide/environment-guide) example) will stop working because of the change.